node/pkg/p2p/p2p.go

```diff
@@ -237,7 +237,7 @@ func Run(obsvC chan *gossipv1.SignedObservation, obsvReqC chan *gossipv1.Observa
 						BootTimestamp: bootTime.UnixNano(),
 					}
 
-					ourAddr := ethcrypto.PubkeyToAddress(gk.PublicKey)
+					ourAddr := ethcrypto.PubkeyToAddress((*guardianSigner).PublicKey())
```

Comment:
We can use the pointer receiver for `ECDSASigner`, then we don't need to deref the pointer, here is a demo: https://github.com/alephium/wormhole-fork/commit/a64b14c54824b9cbda45adfe4c25716a0253fff2

Reply:
Nice, thanks!